docs/reference/mapping/dynamic/field-mapping.asciidoc

```diff
@@ -25,7 +25,7 @@ h| JSON data type h| `dynamic:true` h| `dynamic:runtime`
  |`object`<<dynamic-object-footnote,^1^>>  2*| `object`
  |`array` 2*|  Depends on the first non-`null` value in the array
  |`string` that passes <<date-detection,date detection>> 2*| `date`
- |`string` that passes <<numeric-detection,numeric detection>> | `double` or `long` | `double`
+ |`string` that passes <<numeric-detection,numeric detection>> | `double` or `long` | `float`
```

Comment:
```suggestion
 |`string` that passes <<numeric-detection,numeric detection>> | `float` or `long` | `double`
```

We use float only if the field is indexed, dynamic runtime fields use double by default.

Reply:
Since JSON doesn’t distinguish a double from a float, it will always choose the wider data type. so the suggestion maybe not correct.
1. In field dynamic mapping , JSON float-point value will mapped by elasticsearch float type.
2. Elasticsearch float type index default value is true.

Reply:
That's not how the values should be interpreted. The `float` or long` part refers to the mapping that will be created for `string` that passes <<numeric-detection,numeric detection>>. when `dynamic` is set to true. The last column indicates the mapping type that is used if dynamic is set to `runtime`. 
@javanna @lockewritesdocs should it be `double or long` for runtime though ?

Reply:
yes, it should be `float` or `long` for dynamic:true, and `double` or `long` for dynamic:runtime. Good catch

Reply:
I updated the suggested change accordingly